packages/core/src/components/axes/axis.ts

```diff
@@ -131,16 +148,25 @@ export class Axis extends Component {
 
 		if (axisOptions.scaleType === ScaleTypes.TIME) {
 			if (Tools.getProperty(options, "timeScale", "addSpaceOnEdges")) {
-				// TODO - Need to account for non-day incrementals as well
-				const [startDate, endDate] = domain;
-				startDate.setDate(startDate.getDate() - 1);
-				endDate.setDate(endDate.getDate() + 1);
+				const startDate = new Date(domain[0]);
+				const endDate = new Date(domain[1]);
+				if (differenceInYears(endDate, startDate) > 3) {
```

Comment:
we should move `3` into `configuration.ts` and an interface

Reply:
ok added!